pkg/utils/gardener/shoot_test.go

```diff
@@ -1219,6 +1219,84 @@ var _ = Describe("Shoot", func() {
 				ExtensionsID(extensionsv1alpha1.DNSRecordResource, dnsProviderType2),
 			)))
 		})
+
+		It("should compute the correct list of required extensions (workerless Shoot)", func() {
+			shoot.Spec.Provider.Workers = nil
+
+			result := ComputeRequiredExtensions(shoot, seed, controllerRegistrationList, internalDomain, externalDomain)
+
+			Expect(result).To(Equal(sets.New(
+				ExtensionsID(extensionsv1alpha1.BackupBucketResource, backupProvider),
+				ExtensionsID(extensionsv1alpha1.BackupEntryResource, backupProvider),
+				ExtensionsID(extensionsv1alpha1.ControlPlaneResource, seedProvider),
+				ExtensionsID(extensionsv1alpha1.ControlPlaneResource, shootProvider),
+				ExtensionsID(extensionsv1alpha1.InfrastructureResource, shootProvider),
+				ExtensionsID(extensionsv1alpha1.NetworkResource, networkingType),
+				ExtensionsID(extensionsv1alpha1.WorkerResource, shootProvider),
+				ExtensionsID(extensionsv1alpha1.ExtensionResource, extensionType1),
+				ExtensionsID(extensionsv1alpha1.DNSRecordResource, dnsProviderType1),
+				ExtensionsID(extensionsv1alpha1.DNSRecordResource, dnsProviderType2),
+			)))
+		})
+
+		It("should compute the correct list of required extensions (workerless Shoot with not supported globally enabled extension)", func() {
```

Comment:
This sentence is not clear, it tells the test is for workerless shoot with not supported globally enabled extension, but it rather checks other combinations also.
```suggestion
		It("should compute the correct list of required extensions (workerless Shoot and globally enabled extensions)", func() {
```